### ISSUE TITLE
MD014: False positive when command output follows

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#272]
## Summary

MD014 flags shell commands with `$` prompts even when output is shown after the command.

## Example

```console
$ cargo build
   Compiling myprogram v0.1.0 (file:///projects/myprogram)
warning: variable does not need to be mutable
```

We flag `$ cargo build` but rumdl correctly skips it because output follows.

## Comparison

| Linter | MD014 Violations |
|--------|------------------|
| mdbook-lint | 93 |
| rumdl | 24 |
| markdownlint | 41 |

rumdl description: "Command should show output (add example output or remove $ prompt)"

## Expected Behavior

Only flag `$` commands that have NO output following them in the code block.

## Related

Issue #263 - Linter comparison
